src/client/views/cube-view/cube-view.tsx

```diff
@@ -558,24 +567,78 @@ export class CubeView extends React.Component<CubeViewProps, CubeViewState> {
           </div>
         </div>
 
-        {deviceSize !== "small" ? <ResizeHandle
-          side="right"
-          initialValue={layout.pinboardWidth}
+        {!this.isSmallDevice() && !layout.pinboard.hidden && <ResizeHandle
+          direction={Direction.RIGHT}
+          initialValue={layout.pinboard.width}
           onResize={this.onPinboardPanelResize.bind(this)}
           onResizeEnd={this.onPanelResizeEnd.bind(this)}
           min={MIN_PANEL_WIDTH}
           max={MAX_PANEL_WIDTH}
-        /> : null}
-
-        <PinboardPanel
+        />}
+        {!layout.pinboard.hidden && <PinboardPanel
           style={styles.pinboardPanel}
           clicker={clicker}
           essence={essence}
           timekeeper={timekeeper}
-        />
+        />}
       </div>
       {this.renderRawDataModal()}
       {this.renderViewDefinitionModal()}
     </div>;
   }
+
+  private calculateStyles() {
+    const { layout } = this.state;
+    const isDimensionPanelHidden = layout.factPanel.hidden;
+    const isPinboardHidden = layout.pinboard.hidden;
+    if (this.isSmallDevice()) {
```

Comment:
I would split this into two separate methods - one for small devices and second for others - to increase readability.

Reply:
And reuse few lenses!